src/writer.js

```diff
@@ -0,0 +1,377 @@
+            }, 300);
+          }
+
+          startStop();
+          stopTurns();
+          emptyCardNumber = 0;
+        }
+
+        faceUpCounter = 0;
+        return;
+      }
+      faceUpCard1 = target;
+    }
+  }
+};
+
+export function isFirstBetter (recent, previous) {
+  let recentTime = Number(recent.time.replace(':', ''));
+  let recentRatio = recentTime + recent.turns;
+
+  //make recent better if there is no previous
+  if (!localStorage.getItem(recent.size)) {
+    return 1;
+  }
+
+  let previousTime = Number(JSON.parse(previous).time.replace(':', ''));
+  let previousRatio = previousTime + JSON.parse(previous).turns;
+
+  //if new result is worse than saved one
+  if (recentRatio >= previousRatio) {
+    return 0;
+  }
+
+  //if new result is better than saved one
+  return 1;
+}
+
+export function changeLeadResult (result) {
+  localStorage.removeItem(result.size);
+  document.getElementsByClassName(`size${result.size}`)[0].innerHTML = `<b>Time:</b> ${result.time}, <b>Turns:</b> ${result.turns} `;
+  localStorage.setItem(result.size, JSON.stringify(result));
+}
+
+export function showLeaderboard () {
+  if (localStorage.getItem(44)) {
+    let res1 = JSON.parse(localStorage.getItem(44));
+    document.getElementsByClassName('size44')[0].innerHTML = `<b>Time:</b> ${res1.time}, <b>Turns:</b> ${res1.turns} `;
+  }
+  if (localStorage.getItem(54)) {
+    let res2 = JSON.parse(localStorage.getItem(54));
+    document.getElementsByClassName('size54')[0].innerHTML = `<b>Time:</b> ${res2.time}, <b>Turns:</b> ${res2.turns} `;
+  }
+  if (localStorage.getItem(66)) {
+    let res3 = JSON.parse(localStorage.getItem(66));
+    document.getElementsByClassName('size66')[0].innerHTML = `<b>Time:</b> ${res3.time}, <b>Turns:</b> ${res3.turns} `;
+  }
+}
+
+export function saveGame (time, turns) {
+  let saveTime = document.getElementsByClassName('stopwatch')[0].value;
+  let saveTurns = document.getElementsByClassName('turnsCounter')[0].value;
+  let saveFieldSize = document.getElementsByClassName('fieldSelector')[0].value;
+  let currentField = document.getElementsByClassName('gameField')[0].innerHTML;
+
+  showMsg('<b>Saving...</b>');
+  setTimeout(function () {
+    document.getElementsByClassName('msgDiv')[0].remove();
+  }, 1000);
+
+  clearStopwatch();
+  clearTimeout(clocktimer);
+  init = 0;
+  stopTurns();
+  isFirstClick = true;
+  emptyCardNumber = 0;
+  Array.from(document.getElementsByClassName('card')).forEach((item) => {
+    if (item.classList.contains('invisible')) {
+      emptyCardNumber += 1;
+    }
+  });
+
+  let savedGame = {
+    'size': saveFieldSize,
+    'field': currentField,
+  }
+
+  localStorage.setItem('gameSave', JSON.stringify(savedGame));
+}
+
+export function loadGame (save) {
+  if (!save) {
+    showMsg('<b>No saved games found :(</b>');
+    setTimeout(function () {
+      document.getElementsByClassName('msgDiv')[0].remove();
```

Comment:
maybe you can create message just one and append styles like 'active' or 'hidden'? it requires less performance then create it 20 times :)

Reply:
Seems like there's a huge problem about switching class states because gamefield always changes and either clears its contents or gets removed and put back again (with the hidden messages). So there is a way to create it once and instead of creating it every time I will append and remove it every time. Will it require less performance?

Reply:
@FinnYguY fine

Reply:
@avkulahina I'm just getting more and more hesitated about this suggestion. While I was running after reappending an element, I found myself in creating it's clones. Like you know, .append doesn't simply create new element, it also deletes it on it's previous place, to avoid deleting I need to clone it first, and we just get more lines of code for almost nothing. Does creating affects performance so much? Maybe we can leave the messages the way they are? I mean there aren't a lot of messages and it looks so quick, I even give additional time for the 'Shuffling...' message so the user could understand what's happening rn. 
Compare this:
![image](https://user-images.githubusercontent.com/44834982/89662137-55ac9580-d8dc-11ea-88a7-f31f67ce330d.png)
And this:
![image](https://user-images.githubusercontent.com/44834982/89662225-707f0a00-d8dc-11ea-87e3-9ca2dc786916.png)

Reply:
@FinnYguY you don't need to clone the node at all. the steps are:
- create `Message` block at the app creation/page rendering in hidden mode
- when you need to show the message you can use something like message.show('title', 'message text') which will overwrite **the text** in the block and show the message block
- you can still use the setTimeout function to control message show/hidden state

Reply:
So you mean that I should create the msg-block, append it to the gamefield and then toggle its visibility and text. But as I already mentioned, the gamefield must be changed with every new game and the message will be deleted too. To avoid that I have to clear all the gamefield innerHTML except the msg-block, but (though I dunno how it should be done) I guess that it consumes much more resources to check all the innerHTML and leave msg-block untouched than to just create a div. (Or maybe I just misunderstood you and didn't get what you wanted me to do)

Reply:
No, creation of element is always slower than changing a text node of element. It is not a good practice to remove/add entire block if only one small piece of text is changing. Your app will have one Toast / Alert message. It is not possible you have 2 messages at the same time. Does it correct?

Try to think in component way. There is a component Alert. When you need any of your other components show an alert you need just Alert.showMessage('message text'). Than you expect it will disappear in N ms. It is not import were or what will trigger Alert. Only thing matters - the message.

Reply:
I don't know how and why the container with text depends on fields size. They shouldn't even know about each other.

Also thinking in component way will give you advantage to store actual link to text element.

```
/* Alert.js */
class FancyAlert {
  constructor(){
    this.element = document.createElement('div');
    // some code here
  }

  show(message) {
    /* code here*/
   this.element.innerText = message;
  }
}
const fancyAlert = new Alert();
export default fancyAlert; 

/* Field.js */
import fancyAlert from '/path/to/Alert.js'
....
shuffle () {
  /* some code here */
  fancyAlert.show('Shuffling');
}
```

Reply:
@avkulahina Well, I fully reworked the messages and its appearance, I added a class like you asked, now I just change message text and not creating it again. Please, check out the code again when you have time